server/supernova/sc/sc_ugen_factory.cpp

```diff
@@ -282,11 +282,15 @@ void sc_ugen_factory::load_plugin ( boost::filesystem::path const & path )
 
 void sc_ugen_factory::close_handles(void)
 {
-#if 0
-    /* closing the handles has some unwanted side effects, so we leave them open */
-    for(void * handle : open_handles)
-        dlclose(handle);
-#endif
+    for(void * handle : open_handles){
+        void *ptr = dlsym(handle, "unload");
+        if(ptr){
+            UnLoadPlugInFunc unloadFunc = (UnLoadPlugInFunc)ptr;
+            (*unloadFunc)();
+        }
+        //dlclose(handle);
```

Comment:
not at the moment.
With this commit there are not hangs or crashes.
When it is mature could be tried but the risk is that bad plugin programing in release of resources could made crashes to appear